File: frontend/src/types/project.ts
export interface Project {
  id: string;
  user_id: string;
  name: string;
  description: string | null;
  project_type: string;
  status: string;
  site_count?: number;
  created_at: string;
  updated_at: string;
}

export interface ProjectCreateInput {
  name: string;
  description?: string;
  project_type?: string;
  status?: string;
}

export interface ProjectUpdateInput {
  name?: string;
  description?: string;
  project_type?: string;
  status?: string;
}

File: backend/app/schemas/project.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, Field


class ProjectBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255, description="Project name")
    description: Optional[str] = None
    project_type: str = Field(default="carbon", description="Type of project e.g. carbon, biodiversity")
    status: str = Field(default="active", description="Project status e.g. active, completed, archived")


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    project_type: Optional[str] = None
    status: Optional[str] = None


class ProjectRead(ProjectBase):
    id: UUID
    user_id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
